app/Http/Controllers/AuthController.php

```diff
@@ -572,6 +578,10 @@ public static function verifyCredentials($username, $password)
         }
 
         if ($user != null && Hash::check($password, $user->password)) {
+            if ((new PwnedPasswords())->setPassword($password)->isPwnedPassword() && HashMapItem::where('key', 'pwned-pass')->where('subkey', $user->id)->first() === null) {
```

Comment:
For efficiency, maybe do the HaspMapItem check first before doing the isPwnedPassword-check, as we can do the HasMapItem check locally.

Reply:
Agreed.